### PR TITLE
move MIR syntax into a dedicated file and ping some people whenever it changes

### PR DESCRIPTION
**Pull request by RalfJung**
Adding or changing MIR operations/statements/whatever should be under significant scrutiny wrt their wider impact, specified semantics, and so on. So let's start by putting all that into a dedicated file and pinging some people whenever that file changes.

This PR only moves definitions around, and then fiddles with imports until it all works again.

### CODE REVIEW

**Comment by RalfJung** [comment]
Cc @lcnr @rust-lang/wg-mir-opt let me know if I should add more people (or teams) to the mention list here.

File: triagebot.toml [hunk 1]
```diff
@@ -316,7 +316,12 @@ cc = ["@ehuss"]
...
 [mentions."src/tools/miri"]
+message = "The Miri submodule was changed"
 cc = ["@rust-lang/miri"]
 
 [mentions."src/tools/rustfmt"]
 cc = ["@rust-lang/rustfmt"]
+
+[mentions."compiler/rustc_middle/src/mir/syntax.rs"]
+message = "This PR changes MIR"
+cc = ["@oli-obk", "@RalfJung", "@JakobDegen"]
```
Comment (davidtwco) [inline triagebot.toml:327]:
```suggestion
cc = ["@oli-obk", "@RalfJung", "@JakobDegen", "@davidtwco"]
```
Just to stay in the loop :)

**Comment by oli-obk** [comment]
@bors r+ p=1

let's move this along before it bitrots

**Comment by RalfJung** [comment]
Argh, sorry for that...

**Comment by RalfJung** [comment]
Still some rustdoc stuff...?

**Comment by RalfJung** [comment]
`./x.py doc compiler/rustc_builtin_macros` does not reproduce the error here.

I think I know how to fix this, but how can I check...?

**Comment by RalfJung** [comment]
Oh, we're getting the news twice this time?^^

**Pull request merged**